src/App.jsx

```diff
@@ -13,34 +14,98 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+const SORT_ALPHABET = 'alphabet';
+const SORT_LENGTH = 'length';
+const RESET = 'reset';
+const REVERSE = false;
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+function filteredGoods(goods, filter, reversed, setFilter) {
+  let result = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  if (filter === SORT_ALPHABET) {
+    result.sort((a, b) => a.localeCompare(b));
+  }
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (filter === SORT_LENGTH) {
+    result.sort((a, b) => a.length - b.length);
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (reversed === true) {
+    result = result.reverse();
+  }
+
+  if (filter === RESET) {
+    result = [...goodsFromServer];
+    setFilter('');
+  }
+
+  return result;
+}
+
+export const App = () => {
+  const [filter, setFilter] = useState('');
+  const [reverseToggle, setReverseToggle] = useState(REVERSE);
+
+  function reset() {
+    setFilter(RESET);
+    setReverseToggle(REVERSE);
+  }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const goods = filteredGoods(
+    goodsFromServer,
+    filter,
+    reverseToggle,
+    setFilter,
+  );
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={`button is-info ${filter !== SORT_ALPHABET && 'is-light'}`}
+          onClick={() => setFilter(SORT_ALPHABET)}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={`button is-success ${filter !== SORT_LENGTH && 'is-light'}`}
+          onClick={() => setFilter(SORT_LENGTH)}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={`button is-warning ${!reverseToggle && 'is-light'}`}
+          onClick={() => setReverseToggle(!reverseToggle)}
+        >
+          Reverse
+        </button>
+        {filter || reverseToggle ? (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => reset()}
+          >
+            Reset
+          </button>
+        ) : (
+          ''
+        )}
```

Comment:
```suggestion
        {filter || reverseToggle && (
          <button
            type="button"
            className="button is-danger is-light"
            onClick={() => reset()}
          >
            Reset
          </button>
        )
```
You can simplify, or return null